### PR TITLE
Fix/Simplify regex usage

### PR DESCRIPTION
**Pull request by yschimke**

### CODE REVIEW

**Comment by yschimke** [comment]
Two cases not touched

okhttp/src/main/java/okhttp3/Cookie.kt - reuses matcher
okhttp/src/main/java/okhttp3/MediaType.kt - uses lookingAt instead of find or matchesEntire

**Comment by swankjesse** [comment]
Nice!

File: okcurl/src/main/java/okhttp3/curl/Main.kt [hunk 1]
```diff
@@ -205,7 +205,7 @@ class Main : Runnable {
   private fun mediaType(): MediaType? {
     val mimeType = headers?.let {
       for (header in it) {
-        val parts = header.split(":".toRegex()).toTypedArray()
+        val parts = header.split(':')
```
Comment (JakeWharton) [inline okcurl/src/main/java/okhttp3/curl/Main.kt]: This seems like it should have a limit as well

Comment (swankjesse) [reply okcurl/src/main/java/okhttp3/curl/Main.kt]: https://github.com/square/okhttp/pull/5073

**Pull request merged**